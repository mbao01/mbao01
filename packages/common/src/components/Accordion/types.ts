import * as AccordionPrimitive from "@radix-ui/react-accordion";

export type AccordionProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Root>;

export type AccordionContentProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Content
>;

export type AccordionItemProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Item>;

export type AccordionTriggerProps = React.ComponentPropsWithoutRef<
  typeof AccordionPrimitive.Trigger
>;
