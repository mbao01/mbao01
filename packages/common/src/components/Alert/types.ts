import { type VariantProps } from "../../libs";
import { getAlertClasses } from "./constants";

export type AlertProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getAlertClasses>;

export type AlertTitleProps = React.HTMLAttributes<HTMLHeadingElement>;

export type AlertDescriptionProps = React.HTMLAttributes<HTMLDivElement>;
