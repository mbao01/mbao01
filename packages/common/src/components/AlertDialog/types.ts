import * as AlertDialogPrimitive from "@radix-ui/react-alert-dialog";
import { type VariantProps } from "../../libs";
import { getButtonClasses } from "../Button/constants";
import {
  getDialogContentClasses,
  getDialogDescriptionClasses,
  getDialogFooterClasses,
  getDialogHeaderClasses,
  getDialogOverlayClasses,
  getDialogTitleClasses,
} from "../Dialog/constants";

export type AlertDialogProps = React.ComponentPropsWithoutRef<typeof AlertDialogPrimitive.Root>;

export type AlertDialogOverlayProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Overlay
> &
  VariantProps<typeof getDialogOverlayClasses>;

export type AlertDialogContentProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Content
> &
  VariantProps<typeof getDialogContentClasses>;

export type AlertDialogHeaderProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogHeaderClasses>;

export type AlertDialogFooterProps = React.HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof getDialogFooterClasses>;

export type AlertDialogTitleProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Title
> &
  VariantProps<typeof getDialogTitleClasses>;

export type AlertDialogDescriptionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Description
> &
  VariantProps<typeof getDialogDescriptionClasses>;

export type AlertDialogActionProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;

export type AlertDialogCancelProps = React.ComponentPropsWithoutRef<
  typeof AlertDialogPrimitive.Cancel
> &
  VariantProps<typeof getButtonClasses>;
