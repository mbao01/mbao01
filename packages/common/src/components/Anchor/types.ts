import { VariantProps } from "../../libs";
import { getAnchorClasses } from "./constant";

export type AnchorProps = React.AnchorHTMLAttributes<HTMLAnchorElement> & {
  isExternal?: boolean;
} & VariantProps<typeof getAnchorClasses>;
