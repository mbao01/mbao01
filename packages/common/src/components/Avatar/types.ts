import * as AvatarPrimitive from "@radix-ui/react-avatar";
import { type VariantProps } from "../../libs";
import {
  getAvatarClasses,
  getAvatarFallbackClasses,
  getAvatarImageClasses,
} from "./constants";

export type AvatarProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root>,
  "asChild" | "size"
> &
  VariantProps<typeof getAvatarClasses>;

export type AvatarImageProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>,
  "asChild"
> &
  VariantProps<typeof getAvatarImageClasses>;

export type AvatarFallbackProps = Omit<
  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback>,
  "asChild"
> &
  VariantProps<typeof getAvatarFallbackClasses>;
