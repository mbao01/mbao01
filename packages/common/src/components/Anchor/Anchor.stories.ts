import type { Meta, StoryObj } from "@storybook/react";

import { Anchor } from "./Anchor";

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
  component: Anchor,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
    layout: "centered",
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {},
} satisfies Meta<typeof Anchor>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
  args: {
    href: "/",
    hover: false,
    variant: "default",
    children: "Anchor",
  },
};

export const PrimaryAnchor: Story = {
  args: {
    href: "/",
    variant: "primary",
    children: "Anchor",
  },
};

export const AnchorWithTargetBlank: Story = {
  args: {
    href: "https://example.com",
    target: "_blank",
    children: "InternalWebsite.com",
  },
};

export const ExternalAnchor: Story = {
  args: {
    href: "https://google.com",
    children: "Google.com",
    isExternal: true,
  },
};

export const UnderlineAnchor: Story = {
  args: {
    href: "/",
    underline: true,
    children: "Hover on me",
  },
};

export const HoverAnchor: Story = {
  args: {
    href: "/",
    hover: true,
    children: "Hover on me",
  },
};
