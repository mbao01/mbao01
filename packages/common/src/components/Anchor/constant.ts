import { cva } from "../../libs";

export const getAnchorClasses = cva("link transition-all", {
  variants: {
    variant: {
      accent: "link-accent",
      default: "link-default",
      error: "link-error",
      info: "link-info",
      link: "link-link",
      neutral: "link-neutral",
      primary: "link-primary",
      secondary: "link-secondary",
      success: "link-success",
      warning: "link-warning",
    },
    hover: {
      true: "link-hover",
    },
    underline: {
      false: "no-underline",
    },
  },
});
