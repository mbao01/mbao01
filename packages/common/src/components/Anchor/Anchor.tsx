import { ExternalLinkIcon } from "@radix-ui/react-icons";
import { type AnchorProps } from "./types";
import { getAnchorClasses } from "./constant";
import { cn } from "../../utilities";

export const Anchor = ({
  href,
  hover,
  variant,
  children,
  className,
  underline = false,
  isExternal = false,
  ...props
}: AnchorProps) => {
  return (
    <a
      href={href}
      className={cn(getAnchorClasses({ hover, variant, underline }), className)}
      {...props}
      {...(isExternal && { rel: "noopener noreferrer" })}
    >
      {children}
      {isExternal ? (
        <ExternalLinkIcon name="external" className="ml-[2px] inline" />
      ) : null}
    </a>
  );
};
