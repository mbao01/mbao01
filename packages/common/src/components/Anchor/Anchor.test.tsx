import { render, screen } from "@testing-library/react";
import { Anchor } from "./Anchor";
import { AnchorProps } from "./types";

describe("Anchor", () => {
  const renderAnchor = (props: AnchorProps) => {
    return render(<Anchor {...props} />);
  };

  it("renders a link with correct label", () => {
    const { asFragment } = renderAnchor({
      hover: false,
      children: "Go Home 🏡",
      href: "/home?hello=world#great",
    });

    const anchorEl = screen.getByRole("link", { name: "Go Home 🏡" });

    expect(anchorEl).toBeInTheDocument();
    expect(anchorEl).toHaveAttribute("href", "/home?hello=world#great");
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders an external Anchor", () => {
    const { asFragment } = renderAnchor({
      href: "https://example.com",
      target: "_blank",
      children: "Google.com",
    });

    const anchorEl = screen.getByRole("link", { name: "Google.com" });

    expect(anchorEl).toBeInTheDocument();
    expect(anchorEl).toHaveAttribute("href", "https://example.com");
    expect(asFragment()).toMatchSnapshot();
  });

  it.each([
    "accent",
    "default",
    "error",
    "info",
    "link",
    "neutral",
    "primary",
    "secondary",
    "success",
    "warning",
  ] as const)("has %s variant", (variant) => {
    const label = `${variant} link`;
    const { asFragment } = renderAnchor({
      hover: true,
      children: label,
      href: `/variant/${variant}` as AnchorProps["href"],
    });

    const anchorEl = screen.getByRole("link", { name: label });
    expect(anchorEl).toBeInTheDocument();
    expect(anchorEl).toHaveAttribute("href", `/variant/${variant}`);
    expect(asFragment()).toMatchSnapshot();
  });
});
