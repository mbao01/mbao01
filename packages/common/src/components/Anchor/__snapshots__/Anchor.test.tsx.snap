// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Anchor > has accent variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/accent"
  >
    accent link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has default variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/default"
  >
    default link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has error variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/error"
  >
    error link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has info variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/info"
  >
    info link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has link variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/link"
  >
    link link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has neutral variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/neutral"
  >
    neutral link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has primary variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/primary"
  >
    primary link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has secondary variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/secondary"
  >
    secondary link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has success variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/success"
  >
    success link
  </a>
</DocumentFragment>
`;

exports[`Anchor > has warning variant 1`] = `
<DocumentFragment>
  <a
    class="link transition-all link-hover no-underline"
    href="/variant/warning"
  >
    warning link
  </a>
</DocumentFragment>
`;

exports[`Anchor > renders a link with correct label 1`] = `
<DocumentFragment>
  <a
    class="link transition-all no-underline"
    href="/home?hello=world#great"
  >
    Go Home 🏡
  </a>
</DocumentFragment>
`;

exports[`Anchor > renders an external Anchor 1`] = `
<DocumentFragment>
  <a
    class="link transition-all no-underline"
    href="https://example.com"
    target="_blank"
  >
    Google.com
  </a>
</DocumentFragment>
`;
